snapcraft/commands/init.py

```diff
@@ -16,74 +16,44 @@
-    version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
-    summary: Single-line elevator pitch for your amazing snap # 79 char long summary
-    description: |
-      This is my-snap's description. You have a paragraph or two to tell the
-      most important story about your snap. Keep it under 100 words though,
-      we live in tweetspace and your description wants to look good in the snap
-      store.
 
-    grade: devel # must be 'stable' to release into candidate/stable channels
-    confinement: devmode # use 'strict' once you have the right plugs and slots
+class InitCommand(craft_application.commands.InitCommand):
+    """Snapcraft init command."""
 
-    parts:
-      my-part:
-        # See 'snapcraft plugins'
-        plugin: nil
-    """
-)
+    @override
+    def run(self, parsed_args: argparse.Namespace) -> None:
+        """Pack a directory or run the lifecycle and pack all artifacts."""
+        project_dir = self._get_project_dir(parsed_args)
 
+        if parsed_args.name:
+            craft_cli.emit.progress(
+                "Ignoring '--name' parameter because it is not supported yet.",
+                permanent=True,
+            )
 
-class InitCommand(AppCommand):
-    """Initialize a snapcraft project."""
-
-    name = "init"
-    help_msg = "Initialize a snapcraft project."
-    overview = "Initialize a snapcraft project in the current directory."
-
-    @overrides
-    def run(self, parsed_args):
-        """Initialize a snapcraft project in the current directory.
-
-        :raises SnapcraftError: If a snapcraft.yaml already exists.
-        """
-        emit.progress("Checking for an existing 'snapcraft.yaml'.")
-
-        # if a project is found, then raise an error
         try:
-            project = get_snap_project()
+            craft_cli.emit.progress("Checking for an existing 'snapcraft.yaml'.")
+            project = get_snap_project(project_dir)
             raise errors.SnapcraftError(
-                "could not initialize a new snapcraft project because "
+                "could not initialise a new snapcraft project because "
                 f"{str(project.project_file)!r} already exists"
             )
-        # the `ProjectMissing` error means a new project can be initialized
+            # the `ProjectMissing` error means a new project can be initialized
         except errors.ProjectMissing:
-            emit.progress("Could not find an existing 'snapcraft.yaml'.")
-
-        snapcraft_yaml_path = Path("snap/snapcraft.yaml")
-
-        emit.progress(f"Creating {str(snapcraft_yaml_path)!r}.")
+            craft_cli.emit.progress("Could not find an existing 'snapcraft.yaml'.")
 
-        snapcraft_yaml_path.parent.mkdir(exist_ok=True)
-        snapcraft_yaml_path.write_text(_TEMPLATE_YAML, encoding="utf-8")
+        super().run(parsed_args)
 
-        emit.message(f"Created {str(snapcraft_yaml_path)!r}.")
-        emit.message(
+        craft_cli.emit.message(
```

Comment:
it also feels like maybe this message should be on the initservice; or maybe a way for the service to give a template-based string to be presented to the user after init is done (this would be useful in rockcraft, which has [per-profile](https://github.com/canonical/rockcraft/blob/main/rockcraft/commands/init.py#L387) docs)
(we can also do this later)

Reply:
I moved this to the InitService in snapcraft.

I agree that craft-application should provide a way for applications to give custom messages, but let's do that as a follow up.

Reply:
https://github.com/canonical/craft-application/issues/552